### PR TITLE
Advanced Building Blocks - Bubble Sort

### PR DESCRIPTION
**Pull request by vichuge**
In this project, you will implement the bubble sort algorithm. The assignment is focused on the use of yield and passing a block to a method.

### CODE REVIEW

File: index.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      next if arr[index].length < arr[index + 1].length
+
+      temp = arr[index]
+      arr[index] = arr[index + 1]
+      arr[index + 1] = temp
+    end
+    run -= 1
+  end
+  arr
+end
```
Comment (tirthajyoti-ghosh) [inline index.rb]: Please make use of the `yield` keyword to invoke the block passed to this method  ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_bubble_sort/sortBy.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#assignment-1)).

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## What a remarkable project you have built 💯! This is impressive 🤩👌🧨!

Great job adding the screenshots 👏 .

### But in order to get it approved, we need to work on some changes. Let's see...

- The commit messages can be improved. Use the imperative form to write commit messages.

  Eg.,
  
  Wrong
  ```bash
  adding bubble_sort_by function
  ```

  Right
  ```bash
  Add bubble_sort_by function
  ```

  Notice the first word is capitalized.

  - [x] Please follow the above format when writing commit messages in the future.


- In your README file, please add the steps on how to set up your project locally. Also, add the version of Ruby you are using. This will make your README more professional.

- In the `bubble_sort_by` method, you have not used `yield` keyword. Without that, this method won't work. I have added comments below regarding this.

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Review by Temmarie — CHANGES_REQUESTED** [review]
### Changes Required! :recycle:
Hi @vichuge and @AlejandroNo4  ! 👋 👋 

### You have done well so far :clap: However, you did not finish all the previous review changes. Kindly work on them and submit them for another code review.

**Recommended**

- [ ] bubble_sort_by sorts an array by accepting a block.

```
   Failure/Error: expect(bubble_sort_by(unsorted) { |left, right| left.length - right.length }).to eq %w[hi hey hello]
     
     NoMethodError:
       undefined method `length' for nil:NilClass
```



__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Review by Georjane — APPROVED** [review]
## Status: PR Approved ✔️ ✔️
### Hi Team @vichuge and @AlejandroNo4
Good job on completing this project 🎊🎊🎊 and implementing all changes required by the previous TSE. You can go ahead and merge this PR.

Just a quick notice that it will be nicer if your feature branch in your upcoming projects have more descriptive names. 👍👍

Goodluck on your next project and Happy coding 💻

**Pull request merged**